### DISCUSSION TITLE
Error Using Multiple Poly Files as Mask in Grid_Diag Config File

### DISCUSSION THREAD

**Discussion by chasecalkins** [dtcenter/METplus#1090] [File I/O] [answered]
I am trying to disregard data for the histogram from an nc file '/hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/20110426_WRF_MAX_24hr_RAINNC.nc', against an observation file '/hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/Rain.nc' in which there is no data over NW CONUS.

From the documentation, it should look something like this?:
```
mask = {
   grid    = [ "FULL" ];
   poly    = [ "MET_BASE/poly/LMV.poly",
               "MET_BASE/out/gen_vx_mask/CONUS_poly.nc",
               "MET_BASE/sample_fcst/2005080700/wrfprs_ruc13_12.tm00_G212 \
               {name = \"TMP\"; level = \"Z2\";} >273"
             ];
```

However when using the GridDiagConfig file and setting:
```
mask = {
   //grid = [];
   poly = ["MET_BASE/poly/CONUS.poly", "/hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/Rain.nc {name = \"MAX_STAGEIV_PRECIP_24hr\"; level = \"(*,*)\";}>=0"];
}
```

[root@affd725c306c 24hr]# ( cd /fsx-rd/MET/met-10.0.0/bin/ && exec ./grid_diag -data /hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/20110426_WRF_MAX_24hr_RAINNC.nc -data /hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/Rain.nc -out /hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/rain_accum_hist.nc -config /fsx-rd/MET/met-10.0.0/data/config/GridDiagConfig_precip )
DEBUG 1: Default Config File: /fsx-rd/MET/share/met/config/GridDiagConfig_default
DEBUG 1: User Config File: /fsx-rd/MET/met-10.0.0/data/config/GridDiagConfig_precip
DEBUG 2: Processing masking regions.
WARNING: 
WARNING: Dictionary::lookup_string() -> string lookup failed for name "mask.poly"
WARNING: 
ERROR  : 
ERROR  : Dictionary::lookup_string() -> lookup failed for name "mask.poly"
ERROR  :

However removing the brackets on poly produces:
[root@affd725c306c 24hr]# ( cd /fsx-rd/MET/met-10.0.0/bin/ && exec ./grid_diag -data /hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/20110426_WRF_MAX_24hr_RAINNC.nc -data /hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/Rain.nc -out /hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/rain_accum_hist.nc -config /fsx-rd/MET/met-10.0.0/data/config/GridDiagConfig_precip )
DEBUG 1: Default Config File: /fsx-rd/MET/share/met/config/GridDiagConfig_default
DEBUG 1: User Config File: /fsx-rd/MET/met-10.0.0/data/config/GridDiagConfig_precip
ERROR  : 
ERROR  : yyerror() -> syntax error in file "/tmp/met_config_161_0"
ERROR  : 
ERROR  :    line   = 62
ERROR  : 
ERROR  :    column = 4
ERROR  : 
ERROR  :    text   = ","
ERROR  : 
ERROR  : 
ERROR  :    poly = "MET_BASE/poly/CONUS.poly", "/hailcast-wrf/MET_verification/pcp_combine/Precip/native/24hr/2011/04/Rain.nc {name = \"MAX_STAGEIV_PRECIP_24hr\"; level = \"(*,*)\";}>=0";
ERROR  : ___^_____________________________________________________________________________________________________________________________________________________________________________
ERROR  : 

So my question is, what is the correct syntax that is needed to combine the two because just having either one by itself will work? But together it has these errors

**Comment by JohnHalleyGotway** [comment, marked as answer]
@chasecalkins I see you have a question about configuring the Grid-Diag tool in MET. Whereas other tools, like Point-Stat and Grid-Stat, support the use of multiple masking regions in a single run, the Grid-Diag tool does not.

Here's the "[mask](https://github.com/dtcenter/MET/blob/6d7b55c25c022f80d847dde0ef661fec22a97cb1/met/data/config/GridDiagConfig_default#L57)" dictionary from the default Grid-Diag configuration file:
```
mask = {
   grid = "";
   poly = "";
}
```
Since these entries are formatted as strings ("") instead of square brackets ([]), they only accept a single entry. The square brackets in MET config files indicate that arrays of multiple values are supported. If left to their default values of empty strings, then the PDFs are computed using data from all points in the verification domain. If a "grid" and/or "poly" spatial region is defined, only those grid points which fall into one/both regions are used.

If you need output computed separately for more than one region, you'll need to run the Grid-Diag tool multiple times.

Hope that helps clarify.
